jsonnet/kube-prometheus/kube-proxy/kube-proxy.libsonnet

```diff
@@ -0,0 +1,48 @@
+local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
+
+{
+  prometheus+:: {
+    serviceKubeProxy:
+      local service = k.core.v1.service;
+      local servicePort = k.core.v1.service.mixin.spec.portsType;
+
+      local kubeServicePort = servicePort.newNamed('http-metrics', 10249, 10249);
+
+      service.new('kube-proxy', null, kubeServicePort) +
+      service.mixin.metadata.withNamespace('kube-system') +
+      service.mixin.metadata.withLabels({ 'k8s-app': 'kube-proxy' }) +
+      service.mixin.spec.withClusterIp('None') +
+      service.mixin.spec.withSelector({ 'k8s-app': 'kube-proxy' }),
+    serviceMonitorKubeProxy:
```

Comment:
This seems like a perfect candidate for a `PodMonitor` instead of a `ServiceMonitor`, as there is otherwise kind of a chicken and egg problem with kube-proxy and Services.

Reply:
Did it like [here](https://github.com/coreos/kube-prometheus/blob/master/jsonnet/kube-prometheus/prometheus/prometheus.libsonnet#L234)

System component too, what is difference?

Reply:
Scheduler and controller manager should both be using PodMonitors as well, PodMonitors just didn’t exist when we created those :)

Reply:
Oh, so, ill refactor this too

Reply:
that'd be awesome, but let's keep it separate PRs